src/mono/browser/runtime/marshal-to-cs.ts

```diff
@@ -309,13 +309,22 @@ export class PromiseHolder extends ManagedObject {
 }
 
 function _marshal_task_to_cs(arg: JSMarshalerArgument, value: Promise<any>, _?: MarshalerType, res_converter?: MarshalerToCs) {
+    const handleIsPreallocated = get_arg_type(arg) == MarshalerType.TaskPreCreated;
     if (value === null || value === undefined) {
-        set_arg_type(arg, MarshalerType.None);
-        return;
+        if (MonoWasmThreads && handleIsPreallocated) {
+            // This is multi-threading return from JSImport with Task result and we can't return synchronously,
+            // because C# caller could be on different thread and sent us an async message.
+            // It already returned pending Task to it's own caller.
+            throw new Error("InvalidOperationException: Task return with null value is not supported in multi-threading scenario.");
```

Comment:
it might make sense to push this error into the preallocated handle, so that it's easier to track it back to the call that caused it